components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/internal/OAuth2ServiceComponent.java

```diff
@@ -420,6 +425,29 @@ protected void unsetTokenBinderInfo(TokenBinderInfo tokenBinderInfo) {
         }
     }
 
+    @Reference(name = "response.type.request.validator",
+            service = ResponseTypeRequestValidator.class,
+            cardinality = ReferenceCardinality.MULTIPLE,
+            policy = ReferencePolicy.DYNAMIC,
+            unbind = "unsetResponseTypeRequestValidator")
+    protected void setResponseTypeRequestValidator(ResponseTypeRequestValidator validator) {
+
+        if (log.isDebugEnabled()) {
+            log.debug("Setting the response type request validator for: " + validator.getResponseType());
+        }
```

Comment:
Shall we add the debug log after setting the validator. That way we can know the validator was successfully set. Lets do the same for unsetting.